internal/datacoord/channel_manager.go

```diff
@@ -98,14 +112,19 @@ func NewChannelManager(
 }
 
 // Startup adjusts the channel store according to current cluster states.
-func (c *ChannelManager) Startup(nodes []int64) error {
+func (c *ChannelManager) Startup(ctx context.Context, nodes []int64) error {
 	channels := c.store.GetNodesChannels()
 	// Retrieve the current old nodes.
 	oNodes := make([]int64, 0, len(channels))
 	for _, c := range channels {
 		oNodes = append(oNodes, c.NodeID)
 	}
 
+	// Process watch states for old nodes.
+	if err := c.checkOldNodes(oNodes); err != nil {
```

Comment:
what happened if checkOldNodes is in ChannelWatchState_ToWatch but during the init datanode change it ChannelWatchState_WatchFailure or WatchSuccess before we watch?

Should we start stateChecker before checkOldNodes but add a lock to hold stateChecker?